source/MaterialXGenShader/Nodes/BlurNode.cpp

```diff
@@ -128,41 +145,35 @@ void BlurNode::emitFunctionCall(const ShaderNode& node, GenContext& context, Sha
 
         if (sampleCount > 1)
         {
-            const string MX_MAX_SAMPLE_COUNT_STRING("MX_MAX_SAMPLE_COUNT");
             const string MX_WEIGHT_ARRAY_SIZE_STRING("MX_WEIGHT_ARRAY_SIZE");
             const string MX_CONVOLUTION_PREFIX_STRING("mx_convolution_");
             const string SAMPLES_POSTFIX_STRING("_samples");
             const string WEIGHT_POSTFIX_STRING("_weights");
 
             // Set up sample array
             string sampleName(output->getVariable() + SAMPLES_POSTFIX_STRING);
-            shadergen.emitLine(inputTypeString + " " + sampleName + "[" + MX_MAX_SAMPLE_COUNT_STRING + "]", stage);
-            for (unsigned int i = 0; i < sampleCount; i++)
-            {
-                shadergen.emitLine(sampleName + "[" + std::to_string(i) + "] = " + sampleStrings[i], stage);
-            }
-
+            outputSampleArray(shadergen, stage, inputTypeString, sampleName, sampleStrings);
+           
             // Emit code to evaluate using input sample and weight arrays. 
             // The function to call depends on input type.
             //
             shadergen.emitLineBegin(stage);
             shadergen.emitOutput(output, true, false, context, stage);
             shadergen.emitLineEnd(stage);
 
+            // Emit branched code to compute result baed on filte type
+            //
             shadergen.emitLineBegin(stage);
             shadergen.emitString("if (", stage);
-            // If strings are support compare against string input,
-            // other use int compare
-            if (shadergen.getSyntax().typeSupported(Type::STRING))
-            {
-                shadergen.emitInput(filterTypeInput, context, stage);
-                shadergen.emitString(" == \"" + GAUSSIAN_FILTER + "\")", stage);
-            }
-            else
+            shadergen.emitInput(filterTypeInput, context, stage);
+            // Remap enumeration for comparison as needed
+            std::pair<const TypeDesc*, ValuePtr> result;
+            string emitValue = "\"" + GAUSSIAN_FILTER + "\"";
+            if (syntax.remapEnumeration(GAUSSIAN_FILTER, Type::STRING, FILTER_LIST, result))
```

Comment:
New code to remap filter enum properly. Also get's rid of the magic number "1" comparison for GLSL to a more generic "remap" call.